code/globalincs/pstypes.h

```diff
@@ -308,6 +308,20 @@ constexpr bool LoggingEnabled = false;
 		ASSUME( expr );\
 	} while (false)
 #endif
+
+template <typename T>
+bool CallAssert(bool val, const char *msg, const char *filename, int linenum, T assertMsgFunc)
+{
+	if (!val)
+		assertMsgFunc(msg, filename, linenum, nullptr);
+	ASSUME(val);
```

Comment:
I might be misunderstanding, but will `ASSUME` do what you want here? Since `val` is not modified here, if the compiler assumes it is true, then the previous if statement would never be true, and therefore optimized out, no?

Reply:
Uh, possibly.  To be honest, my sole motivation here was to create a version of `Assert` that could be used in an expression, rather than as a statement.  So I just copied the existing `Assert` and changed as little as possible.  You're probably right, but in that case, the same could be said about the original `Assert` functionality a few lines up.

Interestingly, the `ASSUME` macro is `#define`d to nothing in msvc, clang, gcc, and mingw.  I had thought (ASSUMEd) that it was equivalent to assert on certain platforms, but possibly not.

Reply:
Ah, you are right, it is the same as the above, in which case I'd agree, let's just have consistent behavior.